gateway.go

```diff
@@ -192,27 +179,44 @@ func (gw *Gateway) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Ms
 
 		} else {
 
-			m.Answer = gw.A(state.Name(), ipv4Addrs)
+			m.Answer = gw.A(state.Name(), addrs["A"])
 		}
+
 	case dns.TypeAAAA:
 
-		if len(ipv6Addrs) == 0 {
+		if len(addrs["AAAA"]) == 0 {
 
 			if !isRootZoneQuery {
 				// No match, return NXDOMAIN
 				m.Rcode = dns.RcodeNameError
 			}
 
 			// as per rfc4074 #3
-			if len(ipv4Addrs) > 0 {
+			if len(addrs["A"]) > 0 {
 				m.Rcode = dns.RcodeSuccess
 			}
 
 			m.Ns = []dns.RR{gw.soa(state)}
 
 		} else {
 
-			m.Answer = gw.AAAA(state.Name(), ipv6Addrs)
+			m.Answer = gw.AAAA(state.Name(), addrs["AAAA"])
```

Comment:
What are you doing!?

Reply:
See https://github.com/k8s-gateway/k8s_gateway/pull/101#discussion_r2186075466